docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,105 @@
+- We do not assume that client workloads will be collocated on the same Kubernetes cluster with the data module, 
+furthermore they can run out of Kubernetes at all. 
+  - However, we do check the collocation of a Fybrik module and a relevant workload in the same cluster as a specific use case. 
+- We should allow workload client connections from different IP addresses. 
+
+
+## Network Policies
+
+Fybrik is a Kubernetes application, therefore, the simplest isolation method can be based on 
+[Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) (NP).
+NP allow to control traffic flow at the IP address or port level (OSI layer 3 or 4). If a user workload or a Fybrik 
+previous chained module collocated with the next module in the same Kubernetes cluster, NPs can help to restrict the Fybrik 
+module ingress access. However, when a workload or a previous chained module runs out of the cluster, the NP based 
+isolation is going to be more complicated. The collocated deployment can be based on Kubernetes labels or namespaces, but 
+cross-clusters connections can be based on requester IP addresses, which is not always easy to discover. 
+Therefore, later we will provide other isolation methods as an extension to NP.
+
+### Advantages:
+- does not require any changes in modules implementations, can be implemented in the Fybrik control plane only.
+- can be combined with other isolation methods
+
+### Disadvantages:
+- complicated in cross clusters deployments.
+- NP are implemented by the network plugin. Creating a NetworkPolicy resource without a controller - a Container Network 
+Interface (CNI) that implements it will have no effect. Default Kind deployment which by default uses "kindnetd” doesn’t 
+support NP. It requires installation another CNI, e.g. [Calico](https://github.com/projectcalico/calico). Here is the 
+[instructions](https://alexbrand.dev/post/creating-a-kind-cluster-with-calico-networking/). All K8s clusters in 
+production deployments use CNIs which support NP.
+- requires information about workload deployment, and/or setting labels to the workload
+
+## Implementation
+
+The “from” element of ingress 
+[NetworkPolicySpec](https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/#NetworkPolicySpec) 
+allows to restrict incoming network traffic based on “podSelectors” (labels based), and/or 
+“namespaceSelectors” (labels based) and/or “ipBlocks”. The last one permits incoming network traffic from predefined IP 
+address blocks.
+
+Current “FybrikApplicationSpec” has the “[selector](https://fybrik.io/v1.2/reference/crds/#fybrikapplicationspecselector)” 
+element, which is a combination of “clusterName” and “workloadSelector”. Unfortunately, based on this information we can 
+implement only one label based “podSelector”. Furthermore, for the backward compatibility of the current implementation 
```

Comment:
Why is one label a problem?

Reply:
it is not a one label, but a single label-based selector. (Will connect label and based). In theory you can create a fybrik data plane, and allow different workloads with different labels to connect to the data plane. In addition you cannot restrict access based on namespaces or IP addresses.